nym-node-status-api/nym-node-status-api/src/monitor/mod.rs

```diff
@@ -70,345 +85,342 @@ pub(crate) async fn spawn_in_background(
-        tracing::debug!("{}", gw.ed25519_identity_key().to_base58_string());
-    }
+impl Monitor {
+    async fn run(&mut self) -> anyhow::Result<()> {
+        let default_api_url = self
+            .network_details
+            .endpoints
+            .first()
+            .expect("rust sdk mainnet default incorrectly configured")
+            .api_url()
+            .clone()
+            .expect("rust sdk mainnet default missing api_url");
+
+        let api_client =
+            NymApiClient::new_with_timeout(default_api_url, self.nym_api_client_timeout);
+
+        let all_nodes = api_client
+            .get_all_described_nodes()
+            .await
+            .log_error("get_all_described_nodes")?;
+        tracing::debug!("Fetched {} total nodes", all_nodes.len());
 
-    let mixnodes = all_nodes
-        .iter()
-        .filter(|node| node.description.declared_role.mixnode)
-        .collect::<Vec<_>>();
-    tracing::debug!("Of those, {} mixnodes", mixnodes.len());
+        let gateways = all_nodes
+            .iter()
+            .filter(|node| node.description.declared_role.entry)
+            .collect::<Vec<_>>();
+        tracing::debug!("Of those, {} gateways", gateways.len());
 
-    log_gw_in_explorer_not_api(explorer_gateways.as_slice(), gateways.as_slice());
+        let mixnodes = all_nodes
+            .iter()
+            .filter(|node| node.description.declared_role.mixnode)
+            .collect::<Vec<_>>();
+        tracing::debug!("Of those, {} mixnodes", mixnodes.len());
+
+        let bonded_node_info = api_client
+            .get_all_bonded_nym_nodes()
+            .await?
+            .into_iter()
+            .map(|node| (node.bond_information.node_id, node.bond_information))
+            // for faster reads
+            .collect::<HashMap<_, _>>();
+
+        let mut gateway_geodata = Vec::new();
+        for gateway in gateways.iter() {
+            if let Some(node_info) = bonded_node_info.get(&gateway.node_id) {
+                let gw_geodata = GatewayGeoData {
+                    identity_key: node_info.node.identity_key.to_owned(),
+                    owner: node_info.owner.to_owned(),
+                    pledge_amount: node_info.original_pledge.to_owned(),
+                    location: self.location_cached(&gateway).await,
+                };
+                gateway_geodata.push(gw_geodata);
+            }
+        }
 
-    let all_skimmed_nodes = api_client
-        .get_all_basic_nodes(None)
-        .await
-        .log_error("get_all_basic_nodes")?;
+        // contains performance data
+        let all_skimmed_nodes = api_client
+            .get_all_basic_nodes(None)
+            .await
+            .log_error("get_all_basic_nodes")?;
 
-    let mixnodes = api_client
-        .get_cached_mixnodes()
-        .await
-        .log_error("get_cached_mixnodes")?;
-    tracing::debug!("Fetched {} mixnodes", mixnodes.len());
+        let gateways_blacklisted = all_skimmed_nodes
```

Comment:
I know, I know, you're just replicating old apis, so don't act on this comment, I just have to express my feelings somewhere: it makes no sense to redo mixnode/gateway blacklists anymore. especially since we explicitly show performance and probe scores. @mmsinclair should just remove that column from harbourmaster

Reply:
I'm acting on it indeed, just not in this PR. Upcoming PR will remove those stats in favor of what we agreed was sensible

one thing at a time :wink: